src/Solana/useSolana.tsx

```diff
@@ -0,0 +1,390 @@
+    if (!wallet) return
+    const tokenAccounts = await connection.getTokenAccountsByOwner(wallet, {
+      programId: TOKEN_PROGRAM_ID,
+    })
+
+    const vals = {} as Record<string, bigint>
+    tokenAccounts.value.forEach((tokenAccount) => {
+      const accountData = AccountLayout.decode(tokenAccount.account.data)
+      vals[accountData.mint.toBase58()] = accountData.amount
+    })
+
+    return {
+      hntBalance: vars?.hnt.mint ? vals[vars.hnt.mint] : 0n,
+      iotBalance: vars?.iot.mint ? vals[vars.iot.mint] : 0n,
+      dcBalance: vars?.dc.mint ? vals[vars.dc.mint] : 0n,
+      mobileBalance: vars?.mobile.mint ? vals[vars.mobile.mint] : 0n,
+    }
+  }, [connection, vars, wallet])
+
+  const getSolBalance = useCallback(async () => {
+    if (!wallet) return 0
+    return connection.getBalance(wallet)
+  }, [connection, wallet])
+
+  const submitSolana = useCallback(
+    async ({ txn }: { txn: Buffer }) => {
+      const { txid } = await sendAndConfirmWithRetry(
+        connection,
+        txn,
+        { skipPreflight: true },
+        'confirmed'
+      )
+      return txid
+    },
+    [connection]
+  )
+
+  const submitAllSolana = useCallback(
+    ({ txns }: { txns: Buffer[] }) =>
+      Promise.all(txns.map((txn) => submitSolana({ txn }))),
+    [submitSolana]
+  )
+
+  const getOraclePriceFromSolana = useCallback(
+    async ({
+      tokenType,
+    }: {
+      tokenType: 'HNT'
+    }): Promise<Currency.PriceData | undefined> =>
+      Currency.getOraclePrice({ tokenType, cluster, connection }),
+    [cluster, connection]
+  )
+
+  const createTransferCompressedCollectableTxn = useCallback(
+    async ({
+      collectable,
+      newOwnerHeliumAddress,
+    }: {
+      collectable: Asset
+      newOwnerHeliumAddress: string
+    }): Promise<VersionedTransaction | undefined> => {
+      if (!wallet) return
+      const recipient = heliumAddressToSolPublicKey(newOwnerHeliumAddress)
+      return Hotspot.createTransferCompressedCollectableTxn({
+        collectable,
+        owner: wallet,
+        recipient,
+        connection,
+        url: METAPLEX_URL,
+      })
+    },
+    [connection, wallet]
+  )
+
+  const getHotspots = useCallback(
+    async (
+      opts: Omit<SearchAssetsOpts, 'ownerAddress' | 'creatorAddress'> & {
+        makerName?: string
+      }
+    ) => {
+      if (!wallet) return
+
+      const searchParams = {
+        ownerAddress: wallet.toString(),
+        ...opts,
+      } as SearchAssetsOpts
+
+      if (vars?.hnt.mint) {
+        const hnt = new PublicKey(vars.hnt.mint)
+        const key = entityCreatorKey(daoKey(hnt)[0])[0].toString()
+        searchParams.creatorAddress = key.toString()
+      }
+
+      if (opts.makerName && hemProgram) {
+        const maker = makerKey(opts.makerName)[0]
+        const makerAcc = await hemProgram.account.makerV0.fetch(
+          maker.toString()
+        )
+        searchParams.collection = makerAcc.collection.toString()
```

Comment:
@ChewingGlass Is there a way to pass a list here? Thinking about Makers who have more than one onboarding account.